src/DotNet.Testcontainers/Builders/AbstractBuilder.cs

```diff
@@ -34,6 +35,12 @@ public TBuilderEntity WithDockerEndpoint(string endpoint)
       return this.MergeNewConfiguration(new DockerResourceConfiguration(endpoint: new Uri(endpoint)));
     }
 
+    /// <inheritdoc cref="IAbstractBuilder{TBuilderEntity}" />
+    public TBuilderEntity WithDockerEndpoint(string endpoint, Credentials credentials)
+    {
+      return this.MergeNewConfiguration(new DockerResourceConfiguration(endpoint: new Uri(endpoint), credentials: credentials));
```

Comment:
We should use our own interface. The implementation should consider the common [env variables](https://docs.docker.com/compose/reference/envvars/#docker_host) (`DOCKER_HOST`, `DOCKER_TLS_VERIFY`, `DOCKER_CERT_PATH`) too (at least in the future). Then we can set and apply the default credentials from the constructor based on the machine configuration.

Reply:
If I implement our own interface and class for certificate credentials will it be acceptable for now?
I know that perfect soultion would be connecting to dockerd secured with tls automatically as it is done in java lib but it sounds like a lot of work and as we can see in https://github.com/HofmeisterAn/dotnet-testcontainers/issues/370 last comment was in the end of 2k21 ;)
I'd like to use this lib in my projects but without credentials support is useless now.
We could treat this solution as first step to handle connection to secured dockerd - I don't exclude my further contribution on that

Reply:
> We could treat this solution as first step to handle connection to secured dockerd

I can publish your changes in its own package as soon as I’m back. For develop, I would prefer the mentioned things above.

Reply:
Ok, if I want to try to implement solution with docker env variables, should I do that in this PR or create new one?

Reply:
Yes, a new pull request would be nice. Smaller increments would be appreciated too (first the interface). I'll publish your package later that day.

Reply:
Your changes are published in `1.6.0-beta.GitHub-457`.